File: packages/python/examples/pytest/alert_test.py
def test_alert_accept(al, navigate):
    navigate("alert.html")
    al.do("click the Trigger Alert button")
    al.check("an alert dialog is shown")
    al.do("accept the alert")


def test_confirm_dismiss(al, navigate):
    navigate("alert.html")
    al.do("click the Trigger Confirm button")
    al.check("a confirm dialog is shown")
    al.do("dismiss the alert")
    al.check("the result text says Dismissed")

File: packages/typescript/tests/system/alert.test.ts
import { describe } from "vitest"
import { baseIt } from "./helpers.ts"

describe("Alerts", () => {
  const it = baseIt

  it("accept alert", async ({ expect, setup }) => {
    const { al, $ } = await setup()
    await $.navigate("alert.html")
    await al.do("click the Trigger Alert button")
    await al.check("an alert dialog is shown")
    await al.do("accept the alert")
  })

  it("dismiss confirm", async ({ expect, setup }) => {
    const { al, $ } = await setup()
    await $.navigate("alert.html")
    await al.do("click the Trigger Confirm button")
    await al.check("a confirm dialog is shown")
    await al.do("dismiss the alert")
    await al.check("the result text says Dismissed")
  })
})
